js/packages/openinference-instrumentation-bedrock/src/attributes/request-attributes.ts

```diff
@@ -0,0 +1,402 @@
+/**
+ * Request attribute extraction for AWS Bedrock instrumentation
+ *
+ * Handles extraction of semantic convention attributes from InvokeModel requests including:
+ * - Base model and system attributes
+ * - Input message processing
+ * - Tool definition processing
+ * - Invocation parameters
+ */
+
+import {
+  SemanticConventions,
+  OpenInferenceSpanKind,
+  MimeType,
+  LLMProvider,
+} from "@arizeai/openinference-semantic-conventions";
+import { InvokeModelCommand, InferenceConfiguration } from "@aws-sdk/client-bedrock-runtime";
+import { InvokeModelRequestBody, BedrockMessage } from "../types/bedrock-types";
+import {
+  extractTextFromContent,
+  extractToolResultBlocks,
+  formatImageUrl,
+} from "../utils/content-processing";
+import { setSpanAttribute } from "./attribute-helpers";
+import { Span } from "@opentelemetry/api";
+
+/**
+ * Extracts base request attributes (model, system, provider, invocation parameters)
+ */
+export function extractBaseRequestAttributes(
+  span: Span,
+  command: InvokeModelCommand,
+): void {
```

Comment:
oh one thing i've missed on all these functions typically we prefer "named" parameters over positional since its harder to mess up for other devs. 

like so:


```suggestion
export function extractBaseRequestAttributes({span, command}: {
  span: Span,
  command: InvokeModelCommand,
}): void {
```

for anything with 2 params or more i would say would just go with that format